src/controllers/ratingController.js

```diff
@@ -69,7 +77,44 @@ class ArticleRatingController {
     articleRatingStar = sum / articleRatings.length;
     return articleRatingStar.toFixed(1);
   }
+
+  /**
+   * @description - This method is responsible for calculating average rating of an article
+   *
+   * @static
+   * @param {object} request - Request sent to the router
+   * @param {object} response - Response sent from the controller
+   *
+   * @returns {object} - object representing response message
+   *
+   * @memberof RatingController
+   */
+  static async authorRating(request) {
+    let allArticleRatings;
+    const author = request.params.username;
+    if (author) {
+      allArticleRatings = await Rating.findAll({
+        where: { author }
+      });
+    }
+    if (request.userData) {
```

Comment:
Hi @marcdomain nice implementation so far, can you please explain your thought process here, using two if conditions, won't both of them run nevertheless or is that the plan?

Reply:
The controller was implemented such that it suffices for different cases when you attempt to view user profile: The first caters for a random user viewing someone else's profile. The second caters for a logged in user viewing his/her own profile.
If the different cases were not wrapped in `if block`, the API will throw error (can't find `something...` of undefined). 
So, apparently, the if blocks are very necessary here.

Reply:
Ok then. Good job 👍🏿

Reply:
To add to what Marcus has said, I got an error from our linter while implementing this same approach. Eslint does not allow that because of the no-else-return.

Reply:
yeah @frostyblok that's true, but I was talking about using 'if-else-if' not just 'if-else'. That works fine and makes sure just one of the conditions is executed.